### ISSUE TITLE
Golang support?

### ISSUE THREAD

**Issue by s3rj1k** [lapce/lapce#365]
Does lapce support golang? If so, please describe a needed configuration, thanks.

**Comment by bugadani** [comment]
Hi! "Supports" is an interesting word. You can write go, Lapce supports syntax highlighting, so in a sense, yes, it supports.  We don't have a language server plugin for it, so that support is rather limited.

**Comment by s3rj1k** [comment]
> We don't have a language server plugin for it

Obviously I was referring to language server plugin :), for golang it would be gopls, that is, if VSCode language server protocol is supported.

Has Lapce any knowledge of VSCode language server protocol?

**Comment by bugadani** [comment]
> Has Lapce any knowledge of VSCode language server protocol?

Yes, for example Lapce uses LSP currently to support rust, using rust-analyzer.

We don't have good documentation on how to add language support currently. The best I can do right now is to direct you to [this repository](https://github.com/lapce/lapce-rust) as an example of how the rust support looks like.

**Comment by bugadani** [comment]
On that note, someone sent a message recently on Discors about supporting gopls, so this might change soon enough.

**Comment by nheuillet** [comment]
Hello, a gopls plugin should be out soon enough (less than a week if nothing goes wrong). While still very much in alpha state, most things should work out of the box (completion, hover, format, auto import, etc.)
